src/wallet/mod.rs

```diff
@@ -100,6 +121,45 @@ impl WalletManager {
             .to_owned())
     }
 
+    pub async fn apply_keystream(&self, password: &str, mut data: Vec<u8>) -> anyhow::Result<Vec<u8>> {
+        let mut hasher = VarBlake2b::new(16).unwrap();
+        hasher.update(b"nonce");
+        hasher.update(&password);
+        let mut nonce = [0u8; 16];
+        hasher.finalize_variable(|res| nonce.copy_from_slice(res));
+        let mut salt_str = String::from("");
+        let len = password.len();
+        if len < 4 {
+            salt_str = password.repeat(5-len);
+        }
+        let salt: Salt = Salt::new(&salt_str).unwrap();
```

Comment:
We should probably generate a random salt and save it alongside the encrypted blob.

This makes me think maybe we could rethink the storage of this. Maybe have it as JSON, so that:

1) We can detect if it's encrypted, e.g. look for the `encrypted` key, so it doesn't have to try to read the file twice.
2) We can save the salt in a field.

Example:
```json
{
  "encrypted": "FFA0...",
  "salt": "FFA0..."
}
```

I think there's a way to have `serde` work out which struct to use automatically (via an enum with the structs specified as flattened). If you're keen to do this feel free to go ahead, but I'm happy to do it too.

Reply:
Oh also, in the future we can allow customization of the pbkdf2 params and save them as additional fields.

Reply:
Are you saying transforming `WalletStorage` into an enum with two structs: the current one with the `HashMap<WalletId, Wallet>` and the other a JSON with fields `encrypted` and `salt`?

Reply:
Yup basically 👍